src/scripts/mapping-test-server

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+#
+# Copyright 2014 GoDataDriven B.V.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This is a convenience script for running the avro-tool
+
+# Attempt to set APP_HOME
+# Resolve links: $0 may be a link
+PRG="$0"
+# Need this for relative symlinks.
+while [ -h "$PRG" ] ; do
+    ls=`ls -ld "$PRG"`
+    link=`expr "$ls" : '.*-> \(.*\)$'`
+    if expr "$link" : '/.*' > /dev/null; then
+        PRG="$link"
+    else
+        PRG=`dirname "$PRG"`"/$link"
+    fi
+done
+SAVED="`pwd`"
+cd "`dirname \"$PRG\"`/.." >&-
+APP_HOME="`pwd -P`"
+cd "$SAVED" >&-
+
+
+# Locate config dir and source divolte-env.sh
+[ -n "$DIVOLTE_CONF_DIR" ] && CONF_DIR="$DIVOLTE_CONF_DIR" || CONF_DIR="$APP_HOME/conf"
+
+# add libs to CLASSPATH
+function join { local IFS="$1"; shift; echo "$*"; }
+CLASSPATH=$(join ":" $APP_HOME/lib/*.jar)
+
+# attempt to find java
+if [ -z "$JAVA_HOME" ]; then
+  for candidate in \
```

Comment:
Detecting Java is annoyingly difficult. There are some corner cases here, such as:
 - Do we care which Java 8 version to use, if there are multiple? Right now we'll choose the first, according to lexical sorting, whereas we probably want the latest.
 - Should we really choose Java 8 in preference to whatever `/usr/java/default` is?

We also never verify if the user's `JAVA_HOME` value is correct.

Given this, I'm tempted to propose an alternate strategy:
 - If `JAVA_HOME` is set and valid, use it.
 - If `JAVA_HOME` is not set, java has to be on the path.